spec/services/hashtag_autocomplete_service_spec.rb

```diff
@@ -29,11 +29,16 @@ def self.lookup(guardian_scoped, slugs)
         end
     end
 
-    def self.search(guardian_scoped, term, limit)
-      guardian_scoped
-        .user
-        .bookmarks
-        .where("name ILIKE ?", "%#{term}%")
+    def self.search(guardian_scoped, term, limit, exact_match: false)
```

Comment:
is `exact_match` used in the spec?

More generally, I'm questioning the usefulness of this here? Feels like we are testing some behavior we only have created in this spec?

We should just test a real data source IMO.

Reply:
Yes it is used in the spec, but not directly as an argument the spec passes in -- see the source of `HashtagAutocompleteService#search`. We find exact matches first so they are at the top of the result list then continue searching everything else.

> We should just test a real data source IMO.

I am also testing category + tag data sources here. I added the bookmark one as an example of an additional data source that is not the core category + tag ones, so I have something similar to the channel data source.